src/components/MenuSelector/index.js

```diff
@@ -0,0 +1,181 @@
+    }
+
+    const deleteItem = (id) => {
+        setOrder({
+            ...order,
+            items: order.items.filter(order => order.id !== id )            
+        })
+    }
+
+    const addQuantity = (id, quantity) => {
+        const newItems = order.items.map((item) => {
+            if (item.id === id) {
+                return {
+                    ...item,
+                    quantity,
+                }
+            } else {
+                return item
+            }
+        })
+        setOrder({
+            ...order,
+            items: newItems,
+        })
+    }
+
+    const addTotal = order.items.reduce((result, item) => {
+        return result + item.price * item.quantity
+    }, 0)
+
+    const saveOrder = async() =>{
+        if(order.items.length >0 && order.table <= 0){
+            toast.warn('Por favor asigna la mesa', {
+                className: "rounder-edges",
+                position: "top-center",
+                autoClose: 4000,
+                hideProgressBar: true,
+                transition: Slide
+                });
+        }else if(order.items.length === 0 && order.table >= 1){
+            toast.warn('Agrega al menos un producto a la orden 🍔', {
+                className: "rounder-edges",
+                position: "top-center",
+                autoClose: 4000,
+                hideProgressBar: true,
+                transition: Slide
+                });
+        }else if (order.items.length === 0 && order.table <= 0){
+            toast.warn('Asigna la mesa y productos a la orden 🍔🍟', {
+                className: "rounder-edges",
+                position: "top-center",
+                autoClose: 4000,
+                hideProgressBar: true,
+                transition: Slide
+                });
+        }else{
+            await db
+            .collection('orders')
+            .add(order)
+            setOrder({...initialValues})
+            toast.success('Pedido enviado a cocina! ✨', {
+                className: "rounder-edges",
+                position: "top-center",
+                autoClose: 4000,
+                hideProgressBar: true,
+                transition: Slide
+                });
+        }
+    }
+
+    // prints the lunch components 
+    const restOtdComponents = restOfTheDayData.map(elem =>
+        <MenuElement addProduct={addProduct} data={elem} key={elem.id} />
+    )
+    
+    // prints the breakfast components
+    const breakfastComponents = breakfastData.map(elem =>
+        <MenuElement addProduct={addProduct} data={elem} key={elem.id}/>
+    )
+
+    return(
+    <div className="menu-parent">
+       <div className="menu-container" >
+        <ToggleMenu 
+            className="toggle-btn" 
+            setTab={setTab} 
+            tab={tab}
+        />
+
+        {tab === 'breakfast' ? <div className="items-container">{breakfastComponents}</div> :
+            <div className="items-container">{restOtdComponents}</div>
+        }
+       </div> 
```

Comment:
Acá esamos creando por adelantado ambas colecciones de components `<MenuElement ...>` tanto para `restOtdComponents` como para `breakfastComponents`, sola una colección debería ser creada dependiendo del tab seleccionado por lo cual esta sugerencia que les dejo es más conveniente y con mejor performance.
```suggestion
    return (
      <div className="menu-parent">
        <div className="menu-container">
          <ToggleMenu className="toggle-btn" setTab={setTab} tab={tab} />

          {tab === 'breakfast' ? 
            <div className="items-container">
              {breakfastData.map(elem =>
                <MenuElement addProduct={addProduct} data={elem} key={elem.id}/>
              )}
            </div> :
            <div className="items-container">
              {restOfTheDayData.map(elem =>
                <MenuElement addProduct={addProduct} data={elem} key={elem.id} />
              )}
            </div>
          }
       </div> 
```

Reply:
:0, gracias, no sabía que se estaba creando doble. Lo cambiamos y en próximos proyectos lo tendremos en cuenta :))